### PR TITLE
security integration

### PR DESCRIPTION
**Pull request by AlexKazmirchuk**
added security dependecies, configuration, token filter, some small changes in controllers ...

### CODE REVIEW

File: softgroup/frontend/rest-frontend/src/main/java/com/softgroup/rest/frontend/filters/TokenAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        try {
+            tokenService.validateSessionToken(xToken);
+            String profileID = tokenService.getProfileID(xToken);
+            String deviceID = tokenService.getDeviceID(xToken);
+            SecurityContextHolder
+                    .getContext()
+                    .setAuthentication(new TokenAuthentication(profileID,deviceID));
+            chain.doFilter(req,res);
+        } catch (InvalidTokenException e) {
+            res.getWriter().print("{\n" +
```
Comment (Dominys) [inline softgroup/frontend/rest-frontend/src/main/java/com/softgroup/rest/frontend/filters/TokenAuthenticationFilter.java]: @AlexKazmirchuk Don't do this! Only set status 403

Comment (AlexKazmirchuk) [reply softgroup/frontend/rest-frontend/src/main/java/com/softgroup/rest/frontend/filters/TokenAuthenticationFilter.java]: @Dominys fixed , but if I want custom error response instead standart Tomcat error page ?
(I tried add this:  .exceptionHandling().accessDeniedPage("/403")  to configuration but this  didn`t help)

standart

Comment (Dominys) [reply softgroup/frontend/rest-frontend/src/main/java/com/softgroup/rest/frontend/filters/TokenAuthenticationFilter.java]: @AlexKazmirchuk Why you need some other error? And why you care about error page? Nobody will see that page ))

**Review by Dominys — APPROVED** [review]

File: softgroup/frontend/rest-frontend/src/main/java/com/softgroup/rest/frontend/filters/TokenAuthenticationFilter.java (hunk 1 above)
Comment (AlexKazmirchuk) [reply softgroup/frontend/rest-frontend/src/main/java/com/softgroup/rest/frontend/filters/TokenAuthenticationFilter.java]: @Dominys ok , thanks)

**Pull request merged**